apps/rush-lib/package.json

```diff
@@ -32,6 +32,7 @@
     "glob": "~7.0.5",
     "https-proxy-agent": "~2.2.1",
     "ignore": "~5.1.6",
+    "import-lazy": "~4.0.0",
```

Comment:
@sachinjoseph The lazy-loading optimization you're doing here is very effective.  However it is a bit tricky, particularly since we're working around the lack of `import type`.  It would be a good idea to standardize the code pattern that we use to lazy-load files.  We can document the recommended practice in the API docs for `Import.lazy`.

My proposal:

- Always use `Import.lazy` from `node-core-library` -- no other package.json files should reference `import-lazy` (since it doesn't have TypeScript types).

- Always specify types using `typeof` like this:
    ```ts
    const cliTable: typeof import('cli-table') = Import.lazy('cli-table', require);
    ```
- In cases where the non-runtime types are needed, import them separately using the `Types` suffix:
    ```ts
    const inquirer: typeof import('inquirer') = Import.lazy('inquirer', require);
    import type * as inquirerTypes from 'inquirer';
    ```
- If the compiler does not support `import type` (i.e. until the `downlevel-dts` problem is solved), use a `TODO` comment like this:
    ```ts
    const inquirer: typeof import('inquirer') = Import.lazy('inquirer', require);
    // TODO: Convert this to "import type" after we upgrade to TypeScript 3.8
    import * as inquirerTypes from 'inquirer';
    ```
- If the imported module confusingly has the same name as its export, then use the `Module` suffix:
    ```ts
    const deployManagerModule: typeof import('../../logic/deploy/DeployManager') = Import.lazy(
      '../../logic/deploy/DeployManager',
      require
    );
    // TODO: Convert this to "import type" after we upgrade to TypeScript 3.8
    import * as deployManagerTypes from '../../logic/deploy/DeployManager';
    ```
- If the exports cause a lot of awkwardness (e.g. too many expressions need to have `deployManagerModule.` inserted into them), or if some exports aren't proxiable (e.g. `Import.lazy('thing', require)` returns a function signature), then ***do not lazy-load that module***.  Instead, apply lazy-loading to some other module which is better behaved.  Example: In this PR, **DeployManager.ts** has a bunch of poorly behaved imports -- this was greatly simplified by instead having  **DeployAction.ts** lazy-load **DeployManager.ts**.
- Never apply lazy-loading in a way that would convert the module to `any`, since losing type safety seriously impacts our ability to maintain the code.

Reply:
If you guys are okay with this proposal, I can write up the API docs for `Import`.

Reply:
Sounds good to me.

Reply:
I have changed all the lazy imports to follow the style in this proposal :-)

Reply:
I pushed a commit that adds API docs based on the above.